File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/create_spark_cluster_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.

from .create_cluster_details import CreateClusterDetails
from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CreateSparkClusterDetails(CreateClusterDetails):
    """
    Spark cluster details for creation.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CreateSparkClusterDetails object with values from keyword arguments. The default value of the :py:attr:`~oci.aidataplatform_dp.models.CreateSparkClusterDetails.type` attribute
        of this class is ``USER`` and it should not be changed.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param type:
            The value to assign to the type property of this CreateSparkClusterDetails.
            Allowed values for this property are: "USER", "AGENT_FLOW_COMPUTE"
        :type type: str

        :param display_name:
            The value to assign to the display_name property of this CreateSparkClusterDetails.
        :type display_name: str

        :param description:
            The value to assign to the description property of this CreateSparkClusterDetails.
        :type description: str

        :param driver_config:
            The value to assign to the driver_config property of this CreateSparkClusterDetails.
        :type driver_config: oci.aidataplatform_dp.models.DriverConfig

        :param node_type:
            The value to assign to the node_type property of this CreateSparkClusterDetails.
        :type node_type: str

        :param worker_config:
            The value to assign to the worker_config property of this CreateSparkClusterDetails.
        :type worker_config: oci.aidataplatform_dp.models.WorkerConfig

        :param cluster_runtime_config:
            The value to assign to the cluster_runtime_config property of this CreateSparkClusterDetails.
        :type cluster_runtime_config: oci.aidataplatform_dp.models.ClusterRuntimeConfig

        :param logging_config:
            The value to assign to the logging_config property of this CreateSparkClusterDetails.
        :type logging_config: oci.aidataplatform_dp.models.LoggingConfig

        :param auto_termination_minutes:
            The value to assign to the auto_termination_minutes property of this CreateSparkClusterDetails.
        :type auto_termination_minutes: int

        :param attach_to_notebook_config:
            The value to assign to the attach_to_notebook_config property of this CreateSparkClusterDetails.
        :type attach_to_notebook_config: oci.aidataplatform_dp.models.AttachToNotebookConfig

        :param subscription:
            The value to assign to the subscription property of this CreateSparkClusterDetails.
        :type subscription: oci.aidataplatform_dp.models.SubscriptionDetails

        """
        self.swagger_types = {
            'type': 'str',
            'display_name': 'str',
            'description': 'str',
            'driver_config': 'DriverConfig',
            'node_type': 'str',
            'worker_config': 'WorkerConfig',
            'cluster_runtime_config': 'ClusterRuntimeConfig',
            'logging_config': 'LoggingConfig',
            'auto_termination_minutes': 'int',
            'attach_to_notebook_config': 'AttachToNotebookConfig',
            'subscription': 'SubscriptionDetails'
        }

        self.attribute_map = {
            'type': 'type',
            'display_name': 'displayName',
            'description': 'description',
            'driver_config': 'driverConfig',
            'node_type': 'nodeType',
            'worker_config': 'workerConfig',
            'cluster_runtime_config': 'clusterRuntimeConfig',
            'logging_config': 'loggingConfig',
            'auto_termination_minutes': 'autoTerminationMinutes',
            'attach_to_notebook_config': 'attachToNotebookConfig',
            'subscription': 'subscription'
        }

        self._type = None
        self._display_name = None
        self._description = None
        self._driver_config = None
        self._node_type = None
        self._worker_config = None
        self._cluster_runtime_config = None
        self._logging_config = None
        self._auto_termination_minutes = None
        self._attach_to_notebook_config = None
        self._subscription = None
        self._type = 'USER'

    @property
    def worker_config(self):
        """
        Gets the worker_config of this CreateSparkClusterDetails.

        :return: The worker_config of this CreateSparkClusterDetails.
        :rtype: oci.aidataplatform_dp.models.WorkerConfig
        """
        return self._worker_config

    @worker_config.setter
    def worker_config(self, worker_config):
        """
        Sets the worker_config of this CreateSparkClusterDetails.

        :param worker_config: The worker_config of this CreateSparkClusterDetails.
        :type: oci.aidataplatform_dp.models.WorkerConfig
        """
        self._worker_config = worker_config

    @property
    def cluster_runtime_config(self):
        """
        Gets the cluster_runtime_config of this CreateSparkClusterDetails.

        :return: The cluster_runtime_config of this CreateSparkClusterDetails.
        :rtype: oci.aidataplatform_dp.models.ClusterRuntimeConfig
        """
        return self._cluster_runtime_config

    @cluster_runtime_config.setter
    def cluster_runtime_config(self, cluster_runtime_config):
        """
        Sets the cluster_runtime_config of this CreateSparkClusterDetails.

        :param cluster_runtime_config: The cluster_runtime_config of this CreateSparkClusterDetails.
        :type: oci.aidataplatform_dp.models.ClusterRuntimeConfig
        """
        self._cluster_runtime_config = cluster_runtime_config

    @property
    def logging_config(self):
        """
        Gets the logging_config of this CreateSparkClusterDetails.

        :return: The logging_config of this CreateSparkClusterDetails.
        :rtype: oci.aidataplatform_dp.models.LoggingConfig
        """
        return self._logging_config

    @logging_config.setter
    def logging_config(self, logging_config):
        """
        Sets the logging_config of this CreateSparkClusterDetails.

        :param logging_config: The logging_config of this CreateSparkClusterDetails.
        :type: oci.aidataplatform_dp.models.LoggingConfig
        """
        self._logging_config = logging_config

    @property
    def auto_termination_minutes(self):
        """
        Gets the auto_termination_minutes of this CreateSparkClusterDetails.
        Optional timeout value in minutes used to automatically stop idle compute clusters.


        :return: The auto_termination_minutes of this CreateSparkClusterDetails.
        :rtype: int
        """
        return self._auto_termination_minutes

    @auto_termination_minutes.setter
    def auto_termination_minutes(self, auto_termination_minutes):
        """
        Sets the auto_termination_minutes of this CreateSparkClusterDetails.
        Optional timeout value in minutes used to automatically stop idle compute clusters.


        :param auto_termination_minutes: The auto_termination_minutes of this CreateSparkClusterDetails.
        :type: int
        """
        self._auto_termination_minutes = auto_termination_minutes

    @property
    def attach_to_notebook_config(self):
        """
        Gets the attach_to_notebook_config of this CreateSparkClusterDetails.

        :return: The attach_to_notebook_config of this CreateSparkClusterDetails.
        :rtype: oci.aidataplatform_dp.models.AttachToNotebookConfig
        """
        return self._attach_to_notebook_config

    @attach_to_notebook_config.setter
    def attach_to_notebook_config(self, attach_to_notebook_config):
        """
        Sets the attach_to_notebook_config of this CreateSparkClusterDetails.

        :param attach_to_notebook_config: The attach_to_notebook_config of this CreateSparkClusterDetails.
        :type: oci.aidataplatform_dp.models.AttachToNotebookConfig
        """
        self._attach_to_notebook_config = attach_to_notebook_config

    @property
    def subscription(self):
        """
        Gets the subscription of this CreateSparkClusterDetails.

        :return: The subscription of this CreateSparkClusterDetails.
        :rtype: oci.aidataplatform_dp.models.SubscriptionDetails
        """
        return self._subscription

    @subscription.setter
    def subscription(self, subscription):
        """
        Sets the subscription of this CreateSparkClusterDetails.

        :param subscription: The subscription of this CreateSparkClusterDetails.
        :type: oci.aidataplatform_dp.models.SubscriptionDetails
        """
        self._subscription = subscription

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/CreateSparkClusterDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Spark cluster details for creation.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CreateSparkClusterDetails.Builder.class)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="type")

public final class CreateSparkClusterDetails extends CreateClusterDetails {
    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
        @com.fasterxml.jackson.annotation.JsonProperty("displayName")
        private String displayName;

            public Builder displayName(String displayName) {
            this.displayName = displayName;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("description")
        private String description;

            public Builder description(String description) {
            this.description = description;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("driverConfig")
        private DriverConfig driverConfig;

            public Builder driverConfig(DriverConfig driverConfig) {
            this.driverConfig = driverConfig;
            return this;
            }
        @com.fasterxml.jackson.annotation.JsonProperty("nodeType")
        private String nodeType;

            public Builder nodeType(String nodeType) {
            this.nodeType = nodeType;
            return this;
            }
        
        @com.fasterxml.jackson.annotation.JsonProperty("workerConfig")
        private WorkerConfig workerConfig;

        

        public Builder workerConfig(WorkerConfig workerConfig) {
        this.workerConfig = workerConfig;
        return this;
        }
        
        @com.fasterxml.jackson.annotation.JsonProperty("clusterRuntimeConfig")
        private ClusterRuntimeConfig clusterRuntimeConfig;

        

        public Builder clusterRuntimeConfig(ClusterRuntimeConfig clusterRuntimeConfig) {
        this.clusterRuntimeConfig = clusterRuntimeConfig;
        return this;
        }
        
        @com.fasterxml.jackson.annotation.JsonProperty("loggingConfig")
        private LoggingConfig loggingConfig;

        

        public Builder loggingConfig(LoggingConfig loggingConfig) {
        this.loggingConfig = loggingConfig;
        return this;
        }
            /**
     * Optional timeout value in minutes used to automatically stop idle compute clusters.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("autoTerminationMinutes")
        private Integer autoTerminationMinutes;

                /**
         * Optional timeout value in minutes used to automatically stop idle compute clusters.
         * @param autoTerminationMinutes the value to set
         * @return this builder
         **/
        

        public Builder autoTerminationMinutes(Integer autoTerminationMinutes) {
        this.autoTerminationMinutes = autoTerminationMinutes;
        return this;
        }
        
        @com.fasterxml.jackson.annotation.JsonProperty("attachToNotebookConfig")
        private AttachToNotebookConfig attachToNotebookConfig;

        

        public Builder attachToNotebookConfig(AttachToNotebookConfig attachToNotebookConfig) {
        this.attachToNotebookConfig = attachToNotebookConfig;
        return this;
        }
        
        @com.fasterxml.jackson.annotation.JsonProperty("subscription")
        private SubscriptionDetails subscription;

        

        public Builder subscription(SubscriptionDetails subscription) {
        this.subscription = subscription;
        return this;
        }


        public CreateSparkClusterDetails build() {
            CreateSparkClusterDetails model = new CreateSparkClusterDetails(this.displayName
                    , this.description
                    , this.driverConfig
                    , this.nodeType
                    , this.workerConfig
                    , this.clusterRuntimeConfig
                    , this.loggingConfig
                    , this.autoTerminationMinutes
                    , this.attachToNotebookConfig
                    , this.subscription);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CreateSparkClusterDetails model) {
                this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.driverConfig(model.getDriverConfig());
    this.nodeType(model.getNodeType());
    this.workerConfig(model.getWorkerConfig());
    this.clusterRuntimeConfig(model.getClusterRuntimeConfig());
    this.loggingConfig(model.getLoggingConfig());
    this.autoTerminationMinutes(model.getAutoTerminationMinutes());
    this.attachToNotebookConfig(model.getAttachToNotebookConfig());
    this.subscription(model.getSubscription());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    
    @Deprecated
    public CreateSparkClusterDetails(String displayName, String description, DriverConfig driverConfig, String nodeType, WorkerConfig workerConfig, ClusterRuntimeConfig clusterRuntimeConfig, LoggingConfig loggingConfig, Integer autoTerminationMinutes, AttachToNotebookConfig attachToNotebookConfig, SubscriptionDetails subscription) {
        super(displayName, description, driverConfig, nodeType);
        this.workerConfig = workerConfig;
        this.clusterRuntimeConfig = clusterRuntimeConfig;
        this.loggingConfig = loggingConfig;
        this.autoTerminationMinutes = autoTerminationMinutes;
        this.attachToNotebookConfig = attachToNotebookConfig;
        this.subscription = subscription;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("workerConfig")
    private final WorkerConfig workerConfig;

    
    public WorkerConfig getWorkerConfig() {
        return workerConfig;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("clusterRuntimeConfig")
    private final ClusterRuntimeConfig clusterRuntimeConfig;

    
    public ClusterRuntimeConfig getClusterRuntimeConfig() {
        return clusterRuntimeConfig;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("loggingConfig")
    private final LoggingConfig loggingConfig;

    
    public LoggingConfig getLoggingConfig() {
        return loggingConfig;
    }


        /**
     * Optional timeout value in minutes used to automatically stop idle compute clusters.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("autoTerminationMinutes")
    private final Integer autoTerminationMinutes;

        /**
     * Optional timeout value in minutes used to automatically stop idle compute clusters.
     * @return the value
     **/
    
    public Integer getAutoTerminationMinutes() {
        return autoTerminationMinutes;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("attachToNotebookConfig")
    private final AttachToNotebookConfig attachToNotebookConfig;

    
    public AttachToNotebookConfig getAttachToNotebookConfig() {
        return attachToNotebookConfig;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("subscription")
    private final SubscriptionDetails subscription;

    
    public SubscriptionDetails getSubscription() {
        return subscription;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CreateSparkClusterDetails(");
        sb.append("super=").append(super.toString(includeByteArrayContents));
        sb.append(", workerConfig=").append(String.valueOf(this.workerConfig));
        sb.append(", clusterRuntimeConfig=").append(String.valueOf(this.clusterRuntimeConfig));
        sb.append(", loggingConfig=").append(String.valueOf(this.loggingConfig));
        sb.append(", autoTerminationMinutes=").append(String.valueOf(this.autoTerminationMinutes));
        sb.append(", attachToNotebookConfig=").append(String.valueOf(this.attachToNotebookConfig));
        sb.append(", subscription=").append(String.valueOf(this.subscription));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CreateSparkClusterDetails)) {
            return false;
        }

        CreateSparkClusterDetails other = (CreateSparkClusterDetails) o;
        return java.util.Objects.equals(this.workerConfig, other.workerConfig) &&
            java.util.Objects.equals(this.clusterRuntimeConfig, other.clusterRuntimeConfig) &&
            java.util.Objects.equals(this.loggingConfig, other.loggingConfig) &&
            java.util.Objects.equals(this.autoTerminationMinutes, other.autoTerminationMinutes) &&
            java.util.Objects.equals(this.attachToNotebookConfig, other.attachToNotebookConfig) &&
            java.util.Objects.equals(this.subscription, other.subscription) &&
            super.equals(other);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = super.hashCode();
        result = (result * PRIME) + (this.workerConfig == null ? 43 : this.workerConfig.hashCode());
        result = (result * PRIME) + (this.clusterRuntimeConfig == null ? 43 : this.clusterRuntimeConfig.hashCode());
        result = (result * PRIME) + (this.loggingConfig == null ? 43 : this.loggingConfig.hashCode());
        result = (result * PRIME) + (this.autoTerminationMinutes == null ? 43 : this.autoTerminationMinutes.hashCode());
        result = (result * PRIME) + (this.attachToNotebookConfig == null ? 43 : this.attachToNotebookConfig.hashCode());
        result = (result * PRIME) + (this.subscription == null ? 43 : this.subscription.hashCode());
        return result;
    }


}
